### PR TITLE
Anomaly Neutralizer

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
:cl: Kor
add: Engineering can now create anomaly neutralizer devices, capable of instantly neutralizing the short lived anomalies created by the supermatter engine.
/:cl:

Why: The engine has lots of depth but little reward for learning it. Learning to safely push the engine to the area in which it will create anomalies will now reward engineering, and provide a somewhat reliable way of getting anomaly cores if we expand mechanics around them.

Feels like a natural fit for techwebs.

### CODE REVIEW

**Comment by zaracka** [comment]
thank you for this

**Comment by ThePainkiller** [comment]
I can't wait to work with Syd Echard every shift making a stable anomaly farm by pushing the SM to its limits <3

**Comment by Barhandar** [comment]
s/making a stable anomaly farm/making the server crash yet again/

**Comment by kevinz000** [comment]
now just make tesla armor wearers being shocked trigger their tesla armor too and increase the lightning bolt's damage and pass it on and we'll have some real fun.

File: code/game/objects/items/devices/anomaly_neutralizer.dm [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+	slot_flags = ITEM_SLOT_BELT
+	flags_1 = NOBLUDGEON_1
+
+/obj/item/anomaly_neutralizer/afterattack(atom/target, mob/user, proximity)
+	..()
+	if(!proximity || !target)
+		return
+	if(istype(target, /obj/effect/anomaly))
+		var/obj/effect/anomaly/A = target
+		to_chat(user, "<span class='notice'>[src] crumbles to dust from the strain of neutralizing [A]!</span>")
```
Comment (RandomMarine) [inline code/game/objects/items/devices/anomaly_neutralizer.dm]: Crumbling to dust is something I'd expect from a magic item, not technology.
Maybe re-fluff it to saying it explodes or something.

**Comment by ExcessiveUseOfCobblestone** [comment]
Can we have a separate one that forgo the core for a chunk of techpoints?

File: code/game/objects/items/devices/anomaly_neutralizer.dm (hunk 1 above)
Comment (KorPhaeron) [reply code/game/objects/items/devices/anomaly_neutralizer.dm]: Fair

**Comment by KorPhaeron** [comment]
A way to convert anomalies to tech points should exist if it doesnt already

**Comment by ExcessiveUseOfCobblestone** [comment]
c-c-c-could you turn this into a stack :)?

File: code/game/objects/items/devices/anomaly_neutralizer.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/devices/anomaly_neutralizer.dm]: > crumbles to dust
> isn't deleting

???

Comment (KorPhaeron) [reply code/game/objects/items/devices/anomaly_neutralizer.dm]: its a beta

**Comment by Shadowflame909** [comment]
@kevinz000 So if I run into the tesla engine, and auto clone but my tesla armor stays the same. Will my tesla armor evolve into instant husk armor? Time to set up two engines at one time for the maximum memery. 4 HOUR SERVER MISSION GOALS.

**Comment by optimumtact** [comment]
# YAY

**Review by SpaceManiac — APPROVED** [review]

**Comment by ExcessiveUseOfCobblestone** [comment]
> c-c-c-could you turn this into a stack :)?

**Comment by SpaceManiac** [comment]
You can already fit seven of these in a box, that's more than an entire round's worth unless you're running the SM dangerously, in which case you're standing right next to the protolathe and can print more. If you think it's worth feel free to open another PR.

Design lead approved, code looks good, I've tested it. Merging.

**Pull request merged**